### PR TITLE
update ulab to 5.0.9

### PR DESCRIPTION
**Pull request by jepler**
Closes: #6115

### CODE REVIEW

File: ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk [hunk 1]
```diff
@@ -13,6 +13,8 @@ LONGINT_IMPL = MPZ
 CIRCUITPY_LTO_PARTITION = one
 
 CIRCUITPY_AESIO = 0
+CIRCUITPY_FLOPPYIO = 0
+CIRCUITPY_GIFIO = 0
```
Comment (tannewt) [inline ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk:17]: Is this only gif writing? Seems like folks would want to read it on the matrix portal.

File: ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk [hunk 2]
```diff
@@ -13,6 +13,8 @@ LONGINT_IMPL = MPZ
 CIRCUITPY_LTO_PARTITION = one
 
 CIRCUITPY_AESIO = 0
+CIRCUITPY_FLOPPYIO = 0
```
Comment (tannewt) [inline ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk:16]: Is this on by default? Do we expect it to work on most boards? Maybe it should be off by default.

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for the update! Just a couple questions.

File: ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk (hunk 1 above)
Comment (jepler) [reply ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk:17]: Yes, gifio only writes GIFs.

File: ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk (hunk 2 above)
Comment (jepler) [reply ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk:16]: It is on for all "FULL_BUILD" samd M4 boards (as well as for rp2040).

It should work on any SAM D/E5x board with enough I/Os.

Comment (tannewt) [reply ports/atmel-samd/boards/matrixportal_m4/mpconfigboard.mk:16]: Ok, this is fine for now. If "enough I/Os" are only available on a few boards then we should consider making it opt-in.

**Review by tannewt — APPROVED** [review]

**Pull request merged**